compiler/src/dotty/tools/package.scala

```diff
@@ -38,13 +41,34 @@ package object tools {
 
   def unreachable(x: Any = "<< this case was declared unreachable >>"): Nothing =
     throw new MatchError(x)
+  
 
   transparent inline def assertShort(inline assertion: Boolean, inline message: Any = null): Unit =
     if !assertion then
-      val msg = message
-      val e = if msg == null then AssertionError() else AssertionError("assertion failed: " + msg)
-      e.setStackTrace(Array())
-      throw e
+    summonFrom {
+      case ctx: Context =>  throwAssertionError(message, ctx, truncateStack = true)
+      case _ => throwAssertionError(message, null, truncateStack = true)
+    }
+    
+  transparent inline def assert(inline assertion: Boolean): Unit =
+    assert(assertion, null)
+  
+  transparent inline def assert(inline assertion: Boolean, inline message: Any): Unit =
```

Comment:
It would be better if this one is not called `assert` because it can be confused with the normal assert. There is no way to distinguish a call to this assert without a context with a call to Predef assert.